File: packages/ai_config/domain/provider.py
from typing import Any, Dict, List, Optional

from pydantic import BaseModel

from .auth import AuthSource


class Provider(BaseModel):
    id: str
    name: str
    type: str
    models: Optional[List[str]] = None
    auth: AuthSource
    auth_env_var: Optional[str] = None
    profile_key: Optional[str] = None
    priority: Optional[int] = None
    meta: Optional[Dict[str, Any]] = None

File: packages/ai-config/src/domain/provider.ts
import type { AuthSource } from './auth.js';

export interface Provider {
  id: string;
  name: string;
  type: string;
  models?: string[];
  auth: AuthSource;
  authEnvVar?: string;
  profileKey?: string;
  priority?: number;
  meta?: Record<string, unknown>;
}
